R/read_delim.R

```diff
@@ -125,6 +120,23 @@ NULL
 #' # Or directly from a string with `I()`
 #' read_csv(I("x,y\n1,2\n3,4"))
 #'
+#' # Column selection-----------------------------------------------------------
+#' # Pass column names or indexes directly to select them
+#' read_csv(readr_example("chickens.csv"), col_select = c(chicken, eggs_laid))
+#' read_csv(readr_example("chickens.csv"), col_select = c(1, 3:4))
+#'
+#' # Or use the selection helpers
+#' read_csv(
+#'   readr_example("chickens.csv"),
+#'   col_select = c(starts_with("c"), last_col())
+#' )
+#'
+#' # You can also rename specific columns
+#' read_csv(
+#'   readr_example("chickens.csv"),
+#'   col_select = c(egg_yield = eggs_laid, everything())
+#' )
+#'
```

Comment:
This is adapted from vrooms examples of using `col_select`.

vroom uses the `mtcars.csv` dataset but I noticed that readr has a different version of `mtcars.csv` that didn't work well showcasing `col_select` so I'm using the `chickens.csv` data